packages/deployer/subtasks/deploy-contracts.js

```diff
@@ -62,7 +62,7 @@ async function _processContracts(contracts) {
   const toCreate = [];
 
   for (const [contractName, contractData] of Object.entries(contracts)) {
-    if (hre.network.name === 'hardhat' || !contractData.deployedAddress) {
+    if (!contractData.deployedAddress) {
```

Comment:
There is no need to handle the special case of having the deployment files have something that is not on the blockchain (in this case for the hardhat network, that resets everything on every run). If we want to use the hardhat network outside of tests, we have to just use `--clear` instead.